### ISSUE TITLE
False positive with typing.TypedDict?

### ISSUE THREAD

**Issue by yakMM** [ariebovenberg/slotscheck#120]
Consider the following:

```py
from typing import TypedDict

foo = TypedDict("foo", {"bar": str})


class Foo(TypedDict):
    bar: str
```

slotscheck running in stric mode (`require-subclass = true`) will flag both `foo` and `Foo` as missing slots, although TypedDict are not used as "real class" and don't support slots:
```
PS D:\test> slotscheck main.py -v
ERROR: 'main:Foo' has no slots, but it could have.
       This is flagged because the strict `require-subclass` setting is active.
ERROR: 'main:foo' has no slots, but it could have.
       This is flagged because the strict `require-subclass` setting is active.
Oh no, found some problems!

```

Or should TypedDict support slots out of the box?

**Comment by ariebovenberg** [comment]
@yakMM thanks for reporting, I'll have a look 👀

**Comment by ariebovenberg** [comment]
Did some digging:

```python
>>> foo.__bases__
(<class 'dict'>, )
```

because `dict` has slots, the `require-subclass` setting is activated. However, since typeddicts aren't 'real' classes (`type(foo()) == dict`) there should be a special exception in slotscheck.

**Comment by ItsDrike** [comment]
This should also be ignored for the backported `typing_extensions.TypedDict`, not just for `typing.TypedDict`

**Comment by ariebovenberg** [comment]
@ItsDrike good catch! I'll have a look.

**Comment by ariebovenberg** [comment]
@ItsDrike version 0.16.5 is now out with the fix

**Comment by ItsDrike** [comment]
Thanks!
